src/users/users.gateway.ts

```diff
@@ -5,55 +5,104 @@ import {
   ConnectedSocket,
   WebSocketServer,
   OnGatewayConnection,
+  OnGatewayDisconnect,
+  WsException,
+  BaseWsExceptionFilter,
 } from '@nestjs/websockets';
 import { UsersService } from './users.service';
-import { RegisterUserDto } from './dto/register-user.dto';
-import { Socket } from 'socket.io';
+import { RegisterUserDto as ConnectUserDto } from './dto/register-user.dto';
 import { LoginUserDto } from './dto/login-user.dto';
-import { Logger, UsePipes, ValidationPipe } from '@nestjs/common';
+import {
+  ArgumentsHost,
+  BadRequestException,
+  Catch,
+  Logger,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import { AsyncApiPub, AsyncApiSub } from 'nestjs-asyncapi';
+import { SocketDto } from './dto/socket.dto';
+import { SendJwtDto } from './dto/send-jwt-.dto';
+
+@UsePipes(
+  new ValidationPipe({
+    transform: true,
+    //   exceptionFactory(validationErrors = []) {
+    //     if (this.isDetailedOutputDisabled) {
+    //       return new WsException('Bad request');
+    //     }
+    //     const errors = this.flattenValidationErrors(validationErrors);
 
-@WebSocketGateway()
-export class UsersGateway implements OnGatewayConnection {
-  constructor(
-    private readonly usersService: UsersService,
-    private logger: Logger,
-  ) {}
+    //     return new WsException(errors);
+    //   },
+  }),
+)
+@WebSocketGateway({
+  cors: {
+    origin: '*',
+    allowedHeaders: '*',
+    credentials: true,
+  },
+})
+export class UsersGateway implements OnGatewayDisconnect {
+  private readonly logger = new Logger(UsersGateway.name);
+  constructor(private readonly usersService: UsersService) {}
+
+  handleDisconnect(client: SocketDto) {
+    // this.usersService.safeDisconnect(client);
+  }
 
-  handleConnection() {
-    return 'hello world';
+  /**
+   * connects notLoggedin user on the website to the app
+   * @param connectUserDto the not Loggedin user data needed to connect clients with each other
+   * @param client user socket
+   * @returns void
+   */
+  @SubscribeMessage('connectTonomy')
```

Comment:
suggested API, using verbNoun format:

login - registers/logs in as a did:jwk/antelope
createChannel - creates a channel to another use
sendMessage - sends a message through a channel

does this make sense?

Reply:
the problem is that we will have 2 different connects one that connects a none logged User.
and a one that connects a logged User